git-worktree/src/lib.rs

```diff
@@ -1 +1,127 @@
 #![forbid(unsafe_code, rust_2018_idioms)]
+//! Git Worktree
+
+use git_hash::oid;
+use git_object::bstr::ByteSlice;
+use git_object::BlobRef;
+use quick_error::quick_error;
+use std::convert::TryFrom;
+use std::fs;
+use std::fs::create_dir_all;
+use std::path::Path;
+use std::time::Duration;
+
+#[cfg(unix)]
+use std::os::unix::fs::PermissionsExt;
+
+quick_error! {
+    #[derive(Debug)]
+    pub enum Error {
+        Utf8Error(err: git_object::bstr::Utf8Error) {
+            from()
+            display("Utf8 Error: {}", err)
```

Comment:
This `Display()` implementation is probably worse than having none. The idea is to provide context about what this part of the program tries to do when it happened. The same is true for all following `display(…)` specifications.

Reply:
Should it be something like `"Couldn't convert path to UTF8"` ?